File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/domain/role/RoleRepository.java
package br.com.deltaglobalbank.identity.domain.role;

import java.util.List;
import java.util.Set;
import java.util.UUID;

public interface RoleRepository {
    Role findById(UUID id);

    Role findByCode(RoleCode code);

    List<Role> findAll();

    List<Role> findAllByIds(Set<UUID> ids);

    List<Role> findAllByModuleId(UUID moduleId);

    List<Role> findAllByUserId(UUID userId);

    List<Role> findAllByApiClientId(UUID apiClientId);
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/domain/role/RoleRepository.kt
package br.com.deltaglobalbank.identity.domain.role

import java.util.UUID

interface RoleRepository {
    fun findById(id: UUID): Role?
    fun findByCode(code: RoleCode): Role?
    fun findAll(): List<Role>
    fun findAllByIds(ids: Set<UUID>): List<Role>
    fun findAllByModuleId(moduleId: UUID): List<Role>
    fun findAllByUserId(userId: UUID): List<Role>
    fun findAllByApiClientId(apiClientId: UUID): List<Role>
}
